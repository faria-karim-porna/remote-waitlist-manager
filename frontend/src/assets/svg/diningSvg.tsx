import React from "react";

const DiningSvgComponent = () => {
  return (
    <div>
      <svg xmlns="http://www.w3.org/2000/svg" className="dining-svg">
        <defs></defs>
        <title>Help Center Illustrations</title>
        <g id="sky">
          <rect className="cls-1" />
          <path
            className="cls-2"
            d="M671.22,58.39c-2.54-4.15-8.68-9.84-23.57-13-12.52-2.68-22,3.66-34,5.6C595.76,53.86,581,39.91,566.7,31.45,546.07,19.22,523.19,8.74,498.91,7.32c-12.25-.72-25.15.89-36.51,5.72-10.47,4.44-16.8,14.15-26,20.47C423.92,42.15,409.54,45,394.75,47a33.2,33.2,0,0,0-24.12,16.48H668.28A3.37,3.37,0,0,0,671.22,58.39Z"
          />
          <path
            className="cls-2"
            d="M1288.69,6.57a64.71,64.71,0,0,0-43.47,4.29c-10.63,5-30.07,18.32-28.76,32.33a2.43,2.43,0,0,0,2.43,2.16h141.34q-3.82-4.5-7.75-8.89c-10.22-1.19-19.09-9-27.69-14.13C1313.45,15.61,1301.48,9.9,1288.69,6.57Z"
          />
          <path
            className="cls-2"
            d="M1135.22,71.55c-40.1,16-86.87-26.72-86.87-26.73-29-25.86-73.3-35.11-111.43-28.21-32.24,5.83-51.93,25.52-79.8,39.53-28.43,14.28-53.53-8.34-82.06,5.74A77.27,77.27,0,0,0,743.63,91.6h453.06C1178,54.18,1135.22,71.55,1135.22,71.55Z"
          />
          <path className="cls-3" d="M242.63,186.58h16.51V153.33H242.63Zm2-31.25h12.51v29.25H244.63Z" />
          <path className="cls-3" d="M242.63,272.47h16.51V239.21H242.63Zm2-31.26h12.51v29.26H244.63Z" />
          <path
            className="cls-3"
            d="M212.82,54.37a15.53,15.53,0,0,0-15.51-15.51h-1V72.12h16.51Zm-2,15.75H198.31V40.9a13.53,13.53,0,0,1,12.51,13.47Z"
          />
          <path className="cls-3" d="M242.63,229.53h16.51V196.27H242.63Zm2-31.26h12.51v29.26H244.63Z" />
          <path className="cls-3" d="M242.63,143.64h16.51V110.38H242.63Zm2-31.26h12.51v29.26H244.63Z" />
          <rect className="cls-3" x="244.63" y="241.21" width="12.51" height="29.26" />
          <rect className="cls-3" x="244.63" y="198.27" width="12.51" height="29.26" />
          <path className="cls-3" d="M198.31,40.9V70.12h12.51V54.37A13.53,13.53,0,0,0,198.31,40.9Z" />
          <rect className="cls-3" x="244.63" y="284.16" width="12.51" height="16.12" />
          <rect className="cls-3" x="142.61" y="87.95" width="26.91" height="18.26" />
          <rect className="cls-3" x="142.61" y="108.21" width="26.91" height="18.26" />
          <rect className="cls-3" x="142.61" y="67.7" width="26.91" height="18.26" />
          <rect className="cls-3" x="142.61" y="47.44" width="26.91" height="18.26" />
          <path
            className="cls-3"
            d="M246.11,80.26H185v-2H224.1V62.16a35.37,35.37,0,0,0-35.37-35.37H171.52V300.28h71.11V282.16h16.51v18.12h13V113.45A34.2,34.2,0,0,0,246.11,80.26Zm-49.8-41.4h1a15.53,15.53,0,0,1,15.51,15.51V72.12H196.31Zm62.83,233.61H242.63V239.21h16.51Zm0-42.94H242.63V196.27h16.51Zm0-42.95H242.63V153.33h16.51Zm0-42.94H242.63V110.38h16.51Z"
          />
          <rect className="cls-3" x="142.61" y="26.79" width="26.91" height="18.65" />
          <rect className="cls-3" x="142.61" y="128.46" width="26.91" height="171.81" />
          <rect className="cls-3" x="244.63" y="155.33" width="12.51" height="29.26" />
          <rect className="cls-3" x="244.63" y="112.38" width="12.51" height="29.26" />
          <polygon className="cls-3" points="1372.3 198.82 1332.57 198.82 1332.89 188.08 1372.3 188.08 1372.3 198.82" />
          <polygon className="cls-3" points="1372.3 224.28 1331.81 224.28 1332.13 213.54 1372.3 213.54 1372.3 224.28" />
          <polygon className="cls-3" points="1372.3 186.09 1332.95 186.09 1333.27 175.35 1372.3 175.35 1372.3 186.09" />
          <polygon className="cls-3" points="1372.3 211.56 1332.19 211.56 1332.51 200.81 1372.3 200.81 1372.3 211.56" />
          <polygon className="cls-3" points="1372.3 173.36 1333.33 173.36 1333.65 162.62 1372.3 162.62 1372.3 173.36" />
          <polygon className="cls-3" points="1372.3 160.63 1333.71 160.63 1334.03 149.88 1372.3 149.88 1372.3 160.63" />
          <polygon className="cls-3" points="1372.3 262.48 1330.68 262.48 1331 251.73 1372.3 251.73 1372.3 262.48" />
          <polygon className="cls-3" points="1372.3 275.21 1330.3 275.21 1330.62 264.46 1372.3 264.46 1372.3 275.21" />
          <polygon className="cls-3" points="1372.3 287.94 1329.92 287.94 1330.24 277.19 1372.3 277.19 1372.3 287.94" />
          <polygon className="cls-3" points="1372.3 237.01 1331.43 237.01 1331.75 226.27 1372.3 226.27 1372.3 237.01" />
          <polygon className="cls-3" points="1372.3 147.9 1334.09 147.9 1334.4 137.15 1372.3 137.15 1372.3 147.9" />
          <polygon className="cls-3" points="1372.3 249.75 1331.06 249.75 1331.38 239 1372.3 239 1372.3 249.75" />
          <polygon className="cls-3" points="1374.28 0 1426.91 0 1435.83 300 1374.28 300 1374.28 0" />
          <polygon className="cls-3" points="1372.3 33.33 1337.49 33.33 1337.81 22.58 1372.3 22.58 1372.3 33.33" />
          <polygon className="cls-3" points="1372.3 20.59 1337.87 20.59 1338.19 9.85 1372.3 9.85 1372.3 20.59" />
          <polygon className="cls-3" points="1372.3 46.06 1337.12 46.06 1337.43 35.31 1372.3 35.31 1372.3 46.06" />
          <polygon className="cls-3" points="1372.3 58.79 1336.74 58.79 1337.06 48.04 1372.3 48.04 1372.3 58.79" />
          <polygon className="cls-3" points="1372.3 135.17 1334.46 135.17 1334.78 124.42 1372.3 124.42 1372.3 135.17" />
          <polygon className="cls-3" points="1329.56 300 1329.86 289.92 1372.3 289.92 1372.3 300 1329.56 300" />
          <polygon className="cls-3" points="1372.3 7.86 1338.25 7.86 1338.48 0 1372.3 0 1372.3 7.86" />
          <polygon className="cls-3" points="1372.3 122.44 1334.84 122.44 1335.16 111.69 1372.3 111.69 1372.3 122.44" />
          <polygon className="cls-3" points="1372.3 71.52 1336.36 71.52 1336.68 60.77 1372.3 60.77 1372.3 71.52" />
          <polygon className="cls-3" points="1372.3 109.71 1335.22 109.71 1335.54 98.96 1372.3 98.96 1372.3 109.71" />
          <polygon className="cls-3" points="1372.3 84.25 1335.98 84.25 1336.3 73.5 1372.3 73.5 1372.3 84.25" />
          <polygon className="cls-3" points="1372.3 96.98 1335.6 96.98 1335.92 86.23 1372.3 86.23 1372.3 96.98" />
          <rect className="cls-3" x="580.49" y="207.26" width="22.28" height="18.59" />
          <rect className="cls-3" x="580.49" y="246.89" width="22.28" height="18.59" />
          <rect className="cls-3" x="580.49" y="286.52" width="22.28" height="12.64" />
          <rect className="cls-3" x="580.49" y="167.64" width="22.28" height="18.59" />
          <rect className="cls-3" x="580.49" y="128.01" width="22.28" height="18.59" />
          <rect className="cls-3" x="467.03" y="217.56" width="95.56" height="17.81" />
          <rect className="cls-3" x="467.03" y="197.74" width="95.56" height="17.81" />
          <rect className="cls-3" x="467.03" y="158.12" width="95.56" height="17.81" />
          <rect className="cls-3" x="467.03" y="138.3" width="95.56" height="17.81" />
          <rect className="cls-3" x="467.03" y="177.93" width="95.56" height="17.81" />
          <path
            className="cls-3"
            d="M467,113.86V136.3h97.55V299.16h13.91V284.52h26.28v14.64h14.07V113.86ZM604.77,267.48H578.49V244.89h26.28Zm0-39.63H578.49V205.26h26.28Zm0-39.63H578.49V165.64h26.28Zm0-39.62H578.49V126h26.28Z"
          />
          <rect className="cls-3" x="467.03" y="296.81" width="95.56" height="2.35" />
          <rect className="cls-3" x="467.03" y="257.19" width="95.56" height="17.81" />
          <rect className="cls-3" x="467.03" y="277" width="95.56" height="17.81" />
          <rect className="cls-3" x="467.03" y="237.37" width="95.56" height="17.81" />
          <rect className="cls-3" x="560.4" y="84.44" width="10.56" height="17.54" />
          <path className="cls-3" d="M500.52,73.6v38.26h78V73.6ZM553.06,104H538.5V82.44h14.56Zm19.9,0H558.4V82.44H573Z" />
          <rect className="cls-3" x="540.5" y="84.44" width="10.56" height="17.54" />
          <rect className="cls-3" x="1203.79" y="141.02" width="27.77" height="46.73" />
          <rect className="cls-3" x="1054.21" y="141.02" width="27.77" height="46.73" />
          <rect className="cls-3" x="1203.79" y="221.8" width="27.77" height="46.73" />
          <rect className="cls-3" x="1153.93" y="221.8" width="27.77" height="46.73" />
          <rect className="cls-3" x="1104.07" y="221.8" width="27.77" height="46.73" />
          <rect className="cls-3" x="1153.93" y="141.02" width="27.77" height="46.73" />
          <path
            className="cls-3"
            d="M1030,118.05V300.28h275.72V118.05Zm54,153.85h-31.77V218.44H1084Zm0-80.78h-31.77V137.65H1084Zm49.86,80.78h-31.77V218.44h31.77Zm0-80.78h-31.77V137.65h31.77Zm49.86,80.78h-31.77V218.44h31.77Zm0-80.78h-31.77V137.65h31.77Zm49.86,80.78h-31.77V218.44h31.77Zm0-80.78h-31.77V137.65h31.77Zm49.86,80.78h-31.77V218.44h31.77Zm0-80.78h-31.77V137.65h31.77Z"
          />
          <rect className="cls-3" x="1253.65" y="221.8" width="27.77" height="46.73" />
          <rect className="cls-3" x="1104.07" y="141.02" width="27.77" height="46.73" />
          <rect className="cls-3" x="1054.21" y="221.8" width="27.77" height="46.73" />
          <rect className="cls-3" x="1253.65" y="141.02" width="27.77" height="46.73" />
          <rect className="cls-3" x="396.24" y="6.74" width="52.84" height="18.05" />
          <rect className="cls-3" x="397.24" y="37.09" width="14.93" height="23.86" />
          <rect className="cls-3" x="433.14" y="37.09" width="14.93" height="23.86" />
          <path className="cls-3" d="M388.89,26.79V299.16h67.53V26.79ZM414.17,63H395.24V35.09h18.93Zm35.9,0H431.14V35.09h18.93Z" />
          <rect className="cls-3" x="312.77" y="266.21" width="11.86" height="25.91" transform="translate(-104.05 307.12) rotate(-45)" />
          <polygon className="cls-3" points="323.67 295.62 318.83 300.46 328.51 300.46 323.67 295.62" />
          <path className="cls-3" d="M346.79,256c-.83-10-1-17.19-1-20.58h-3.55l.93.93-8,8Z" />
          <path className="cls-3" d="M312.18,264.26l-8.35-8.35c-.37,4.53-.89,9.62-1.6,15.17l1.57,1.57Z" />
          <polygon className="cls-3" points="306.34 235.44 323.67 252.77 332.06 244.38 323.89 236.22 324.66 235.44 306.34 235.44" />
          <polygon className="cls-3" points="333.61 242.84 341 235.44 326.21 235.44 333.61 242.84" />
          <polygon className="cls-3" points="333.61 285.68 325.22 294.07 331.61 300.46 348.38 300.46 333.61 285.68" />
          <path className="cls-3" d="M348,268.16c-.37-3.07-.68-6-.93-8.74l-13.49-13.49-8.39,8.39,18.32,18.33Z" />
          <path
            className="cls-3"
            d="M322.12,294.07,303.8,275.74l-2.57,2.57c-.15,1-.3,2-.47,3.07-.35,2.26-.74,4.58-1.16,6.94l12.13,12.14h4Z"
          />
          <path className="cls-3" d="M304.06,253l9.67,9.68,8.39-8.39L304.8,237C304.76,240.36,304.59,245.87,304.06,253Z" />
          <path className="cls-3" d="M304.82,235.47l0,0h0Z" />
          <path className="cls-3" d="M301.75,274.69l.5-.49-.38-.38Z" />
          <rect className="cls-3" x="322.71" y="256.27" width="11.86" height="25.91" transform="translate(-94.12 311.24) rotate(-45)" />
          <path className="cls-3" d="M348.37,270.92l-3.28,3.28,4.33,4.32Q348.83,274.59,348.37,270.92Z" />
          <path className="cls-3" d="M350,282.19l-6.44-6.45-8.38,8.39,16.32,16.33h1.93C352,294.08,350.91,288,350,282.19Z" />
          <polygon className="cls-3" points="275.53 197.68 280.7 219.93 282.02 219.93 282.02 197.68 275.53 197.68" />
          <polygon
            className="cls-3"
            points="347.43 35.07 332.37 35.07 332.37 15.07 318.25 15.07 318.25 35.07 303.19 35.07 303.19 46.61 347.43 46.61 347.43 35.07"
          />
          <polygon className="cls-3" points="368.6 197.68 368.6 219.93 370.63 219.93 375.3 197.68 368.6 197.68" />
          <rect className="cls-3" x="284.21" y="197.68" width="11.87" height="22.25" />
          <rect className="cls-3" x="312.34" y="197.68" width="11.87" height="22.25" />
          <rect className="cls-3" x="298.27" y="197.68" width="11.87" height="22.25" />
          <rect className="cls-3" x="326.4" y="197.68" width="11.87" height="22.25" />
          <path className="cls-3" d="M299.12,290.94c-.58,3.09-1.23,6.28-1.93,9.52h11.44Z" />
          <polygon
            className="cls-3"
            points="293.17 233.29 293.17 233.25 367.84 233.25 370.17 222.12 281.21 222.12 283.8 233.29 293.17 233.29"
          />
          <rect className="cls-3" x="340.47" y="197.68" width="11.87" height="22.25" />
          <rect className="cls-3" x="354.53" y="197.68" width="11.87" height="22.25" />
          <polygon
            className="cls-3"
            points="340.61 188.38 309.45 188.38 309.45 188.41 273.37 188.41 275.02 195.49 375.76 195.49 377.24 188.41 340.61 188.41 340.61 188.38"
          />
          <polygon
            className="cls-3"
            points="313.1 60.17 335.36 60.17 335.36 60.09 347.43 60.09 347.43 48.8 303.19 48.8 303.19 60.09 313.1 60.09 313.1 60.17"
          />
          <polygon className="cls-3" points="320.29 185.38 340.48 185.38 335.48 63.17 320.29 63.17 320.29 185.38" />
          <polygon className="cls-3" points="318.29 63.17 313.01 63.17 309.54 185.38 318.29 185.38 318.29 63.17" />
        </g>
        <g id="window_wall" data-name="window/wall">
          <polygon
            className="cls-4"
            points="1467.24 0 1467.24 124.59 1276.62 124.59 1276.62 0 1273.54 0 1273.54 124.59 1085.98 124.59 1085.98 0 1082.9 0 1082.9 124.59 895.36 124.59 895.36 0 892.27 0 892.27 124.59 704.73 124.59 704.73 0 701.64 0 701.64 124.59 514.1 124.59 514.1 0 511.01 0 511.01 124.59 323.46 124.59 323.46 0 320.38 0 320.38 124.59 132.75 124.59 132.75 0 0 0 0 300 132.75 300 132.75 127.59 320.38 127.59 320.38 300 323.46 300 323.46 127.59 511.01 127.59 511.01 300 514.1 300 514.1 127.59 701.64 127.59 701.64 300 704.73 300 704.73 127.59 892.27 127.59 892.27 300 895.36 300 895.36 127.59 1082.9 127.59 1082.9 300 1085.98 300 1085.98 127.59 1273.54 127.59 1273.54 300 1276.62 300 1276.62 127.59 1467.24 127.59 1467.24 300 1600 300 1600 0 1467.24 0"
          />
          <rect className="cls-5" x="110.75" width="3.35" height="202.7" />
          <rect className="cls-5" x="1487.16" width="3.35" height="300" />
          <path className="cls-6" d="M287.77,53.73H171.4s-6.69,116-26.79,149L180.38,300h74.3C263.17,217.38,292.67,137.73,287.77,53.73Z" />
          <polygon className="cls-7" points="0 0 91.12 0 0 300 0 0" />
          <polygon className="cls-7" points="1600 0 1508.88 0 1600 300 1600 0" />
          <path className="cls-8" d="M195.37,226.07c-1.48-13.27-13.56-23.37-28-23.37H0V300H203.62Z" />
          <path className="cls-8" d="M71.19,145.89l-5.08-32.6H41.9l-5.09,32.6a17.4,17.4,0,1,0,34.38,0Z" />
          <polygon
            className="cls-8"
            points="71.68 202.7 54.97 197.64 54.97 165.97 53.03 165.97 53.03 197.94 53.47 197.94 36.33 202.7 71.68 202.7"
          />
          <path
            className="cls-6"
            d="M1312.23,53.73H1428.6s6.69,116,26.79,149L1419.62,300h-74.3C1336.83,217.38,1307.33,137.73,1312.23,53.73Z"
          />
          <path className="cls-8" d="M1404.63,226.07c1.48-13.27,13.56-23.37,28-23.37H1600V300H1396.38Z" />
          <path className="cls-8" d="M1528.81,145.89l5.08-32.6h24.21l5.09,32.6a17.4,17.4,0,1,1-34.38,0Z" />
          <polygon
            className="cls-8"
            points="1528.32 202.7 1545.03 197.64 1545.03 165.97 1546.97 165.97 1546.97 197.94 1546.53 197.94 1563.67 202.7 1528.32 202.7"
          />
        </g>
        <g id="table">
          <path
            className="cls-9"
            d="M484.23,53.73H367.85C363,137.73,392.46,217.38,401,300h74.3L511,202.7C490.92,169.77,484.23,53.73,484.23,53.73Z"
          />
          <path
            className="cls-9"
            d="M1232.15,53.73H1115.77s-6.69,116-26.79,149l35.77,97.3h74.3C1207.54,217.38,1237,137.73,1232.15,53.73Z"
          />
          <path
            className="cls-2"
            d="M1167.81,300l-8.24-73.93c-1.48-13.27-13.57-23.37-28-23.37H468.42c-14.42,0-26.51,10.1-28,23.37L432.19,300Z"
          />
          <path className="cls-10" d="M556.72,202.7c27,0,52.17-5.75,73.12-15.63H483.6C504.55,197,529.68,202.7,556.72,202.7Z" />
          <path className="cls-10" d="M1043.28,202.7c27,0,52.18-5.75,73.12-15.63H970.16C991.11,197,1016.25,202.7,1043.28,202.7Z" />
          <path
            className="cls-2"
            d="M1091.11,162.46a19.06,19.06,0,0,0,17.76-22l-5.57-35.69h-26.51l-5.57,35.69a19.06,19.06,0,0,0,17.76,22v34.25l-14.55,5.95h31.23l-14.55-5.95Z"
          />
          <path className="cls-9" d="M1103.94,123.54h-27.79l-2.62,16.78A16.72,16.72,0,0,0,1090,159.61h0a16.72,16.72,0,0,0,16.52-19.29Z" />
          <path
            className="cls-2"
            d="M528.38,162.46a19,19,0,0,0,17.76-22l-5.57-35.69h-26.5l-5.57,35.69a19.05,19.05,0,0,0,17.76,22v34.25l-14.56,5.95h31.24l-14.56-5.95Z"
          />
          <path className="cls-9" d="M541.21,123.54H513.43l-2.62,16.78a16.72,16.72,0,0,0,16.51,19.29h0a16.71,16.71,0,0,0,16.51-19.29Z" />
          <line className="cls-11" x1="527.32" y1="241.7" x2="527.32" y2="299.82" />
          <line className="cls-11" x1="739.12" y1="243.13" x2="739.12" y2="301.25" />
          <line className="cls-11" x1="1063.96" y1="243.13" x2="1063.96" y2="301.25" />
          <line className="cls-12" x1="544.04" y1="231.8" x2="544.04" y2="301.25" />
          <line className="cls-12" x1="733.26" y1="258.8" x2="733.26" y2="301.25" />
          <line className="cls-12" x1="1070.99" y1="236.55" x2="1070.99" y2="296.97" />
        </g>
      </svg>
    </div>
  );
}

export const DiningSvg = React.memo(DiningSvgComponent);
