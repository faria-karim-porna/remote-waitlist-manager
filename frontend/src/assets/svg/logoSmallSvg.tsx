import React from "react";

const LogoSmallSVGComponent = () => {
  return (
    <div>
      <svg
        version="1.1"
        id="Layer_1"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
        x="0px"
        y="0px"
        viewBox="0 0 58.5 58.5"
        xmlSpace="preserve"
        className="logoSmallSvg"
      >
        <g>
          <path
            className="st1"
            d="M26.5,9.5l-8.9-8.9c-0.3-0.3-0.9-0.4-1.3-0.3C9.1,3.5,3.3,9.3,0.1,16.4c-0.2,0.4-0.1,1,0.3,1.3l8.9,8.9
     c0.3,0.3,0.9,0.2,1-0.3c1.3-8.1,7.7-14.6,15.9-15.9C26.6,10.4,26.8,9.8,26.5,9.5z"
          />
          <path
            className="st1"
            d="M49,26.7l8.9-8.9c0.3-0.3,0.4-0.9,0.3-1.3C55,9.3,49.2,3.5,42.1,0.3c-0.4-0.2-1-0.1-1.3,0.3l-8.9,8.9
     c-0.3,0.3-0.2,0.9,0.3,1c8.1,1.3,14.6,7.7,15.9,15.9C48.1,26.8,48.7,27,49,26.7z"
          />
          <path
            className="st1"
            d="M57.9,41L49,32c-0.3-0.3-0.9-0.2-1,0.3c-1.3,8.1-7.7,14.6-15.9,15.9c-0.5,0.1-0.7,0.7-0.3,1l8.9,8.9
     c0.3,0.3,0.9,0.4,1.3,0.3c7.2-3.2,12.9-9,16.1-16.1C58.4,41.8,58.3,41.3,57.9,41z"
          />
          <path
            className="st1"
            d="M9.3,32L0.4,41c-0.3,0.3-0.4,0.9-0.3,1.3c3.2,7.2,9,12.9,16.1,16.1c0.4,0.2,1,0.1,1.3-0.3l8.9-8.9
     c0.3-0.3,0.2-0.9-0.3-1c-8.1-1.3-14.6-7.7-15.9-15.9C10.2,31.9,9.6,31.7,9.3,32z"
          />
        </g>
      </svg>
    </div>
  );
};

export const LogoSmallSVG = React.memo(LogoSmallSVGComponent);
