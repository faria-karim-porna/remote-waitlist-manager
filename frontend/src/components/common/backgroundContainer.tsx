import React from "react";

type BackgroundContainerProps = {
  imageURL?: string;
  gradientColor?: string;
};

const BackgroundContainerComponent: React.FC<React.PropsWithChildren<BackgroundContainerProps>> = ({ imageURL, gradientColor, children }) => {
  return (
    <div
      className="bg-photo"
      style={{ backgroundImage: imageURL ? `url(${imageURL})` : gradientColor ? `linear-gradient(${gradientColor})` : "" }}
    >
      {children}
    </div>
  );
};

export const BackgroundContainer = React.memo(BackgroundContainerComponent);
