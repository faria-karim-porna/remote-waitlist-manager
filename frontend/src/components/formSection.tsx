import React, { useState } from "react";
import { unwrapResult } from "@reduxjs/toolkit";
import { joinUser } from "./core/redux/apiSlices/userApiSlice";
import { UserAction } from "./core/redux/slices/userSlice";
import { setUserInLocalStorage } from "./storages/localStorage";
import { useAppDispatch } from "./core/redux/store";

const FormSectionComponent = () => {
  const dispatch = useAppDispatch();
  const [name, setName] = useState("");
  const [partySize, setPartySize] = useState<string>("");
  const canSubmit = () => {
    return name && partySize;
  };
  const handleJoin = () => {
    dispatch(joinUser({ name: name, partySize: Number(partySize) }))
      .then(unwrapResult)
      .then((response) => {
        if (response) {
          dispatch(UserAction.setUserInfo(response));
          setUserInLocalStorage(name);
        }
      })
      .catch((error) => {
        if (error) {
          dispatch(UserAction.setErrorMessage("Username already exist. Please choose a different one."));
        }
      });
  };
  return (
    <div className="waitlist-form">
      <div>
        <input
          type="text"
          placeholder="Enter your name"
          value={name}
          onChange={(e) => setName(e.target.value)}
          className="waitlist-form-input"
        />
      </div>
      <div>
        <input
          type="text"
          placeholder="Party size (should be between 1-10)"
          value={partySize}
          onChange={(e) => {
            const value = e.target.value;
            if ((value === "" || /^[0-9]+$/.test(value)) && (value === "" || (parseInt(value, 10) <= 10 && parseInt(value, 10) >= 1))) {
              setPartySize(value);
            } else {
              setPartySize(value.slice(0, -1));
            }
          }}
          className="waitlist-form-input"
        />
      </div>
      <button onClick={() => handleJoin()} className="square-button" disabled={!canSubmit()}>
        Submit
      </button>
    </div>
  );
};

export const FormSection = React.memo(FormSectionComponent);
