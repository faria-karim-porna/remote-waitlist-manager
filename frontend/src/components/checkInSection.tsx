import React from "react";
import { useAppDispatch } from "./core/redux/store";
import { checkInUser } from "./core/redux/apiSlices/userApiSlice";
import { unwrapResult } from "@reduxjs/toolkit";
import { UserAction } from "./core/redux/slices/userSlice";
import { FoodSvg } from "../assets/svg/foodSvg";

type CheckInSectionProps = {
  name?: string;
};

const CheckInSectionComponent = (props: CheckInSectionProps) => {
  const { name } = props;
  const dispatch = useAppDispatch();

  const handleCheckIn = (name: string) => {
    dispatch(checkInUser(name))
      .then(unwrapResult)
      .then((response) => {
        if (response) {
          dispatch(UserAction.setUserInfo(response));
        }
      });
  };
  return (
    <div>
      <div className="font-25 font-primary font-weight-500">Your table is just a check-in away. Tap to get started!</div>
      <div>
        <div>
          <div className="d-flex justify-content-center">
            <FoodSvg />
          </div>
          <div>
            <button onClick={() => handleCheckIn(name ?? "")} className="square-button">
              Check in
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export const CheckInSection = React.memo(CheckInSectionComponent);
