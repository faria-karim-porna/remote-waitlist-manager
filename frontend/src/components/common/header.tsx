import React from "react";
import { LogoSVG } from "../../assets/svg/logoSvg";

const HeaderComponent = () => {
  return (
    <div className="header w-100 d-flex justify-content-center align-items-center fixed-top">
      <div className="d-flex w-75 align-items-center position-relative">
        <LogoSVG />
      </div>
    </div>
  );
};

export const Header = React.memo(HeaderComponent);
