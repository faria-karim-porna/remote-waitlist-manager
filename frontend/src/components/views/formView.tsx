import React from "react";
import { BackgroundContainer } from "../common/backgroundContainer";
import bgPhoto from "../../assets/images/formViewBg.png";
import { FormSection } from "../formSection";

const FormViewComponent = () => {
  return (
    <BackgroundContainer imageURL={bgPhoto}>
      <div className="d-flex justify-content-center h-100">
        <div className="w-75 d-flex align-items-center h-100">
          <div className="form-view-main">
            <div>
              <div className="font-40 font-weight-700">Book amazing restaurants</div>
              <div className="my-4">
                <div className="font-22 font-primary">
                  From local favorites to the trending restaurants, enjoy without the wait. Discover popular restaurants where you can skip
                  the wait by booking through TableCheck.
                </div>
              </div>
            </div>
            <FormSection />
          </div>
        </div>
      </div>
    </BackgroundContainer>
  );
};

export const FormView = React.memo(FormViewComponent);
