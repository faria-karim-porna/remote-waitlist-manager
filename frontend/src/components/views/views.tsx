import React from "react";
import { useAppSelector } from "../core/redux/store";
import { shallowEqual } from "react-redux";
import { LoadingView } from "./loadingView";
import { FormView } from "./formView";
import { WaitListView } from "./waitListView";
import { DineInView } from "./dineInView";
import { RejoinView } from "./rejoinView";
import { EnumStatus } from "../core/dataTypes/enums/userEnum";

const ViewsComponent = () => {
  const store = useAppSelector(
    (state) => ({
      isBusy: state.userApi.userFetch.isBusy,
      user: state.user.userInfo,
    }),
    shallowEqual
  );

  return (
    <>
      {store.isBusy ? (
        <LoadingView />
      ) : !store.user?.name ? (
        <FormView />
      ) : store.user?.status === EnumStatus.InWaitingList ? (
        <WaitListView />
      ) : store.user?.status === EnumStatus.SeatIn ? (
        <DineInView />
      ) : store.user?.status === EnumStatus.ServiceCompleted ? (
        <RejoinView />
      ) : null}
    </>
  );
};

export const Views = React.memo(ViewsComponent);
