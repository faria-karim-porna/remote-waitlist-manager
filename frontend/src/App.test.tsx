import { render, screen } from "@testing-library/react";
import { Header } from "./components/common/header";
import { LogoSVG } from "./assets/svg/logoSvg";
import { FormView } from "./components/views/formView";
import { BackgroundContainer } from "./components/common/backgroundContainer";
import bgPhoto from "./assets/images/formViewBg.png";
import { FormSection } from "./components/formSection";

jest.mock("./assets/svg/logoSvg", () => ({
  LogoSVG: jest.fn(() => <svg data-testid="logo-svg">Mock Logo</svg>),
}));

jest.mock("./components/common/backgroundContainer", () => ({
  BackgroundContainer: jest.fn(({ children }) => <div data-testid="background-container">{children}</div>),
}));

jest.mock("./components/formSection", () => ({
  FormSection: jest.fn(() => <div data-testid="form-section">Mock FormSection</div>),
}));

describe("Header Component", () => {
  it("renders the header with the correct structure and classes", () => {
    const { container } = render(<Header />);
    const headerDiv = container.querySelector(".header");

    expect(headerDiv).toBeInTheDocument();
    expect(headerDiv).toHaveClass("w-100 d-flex justify-content-center align-items-center fixed-top");
  });

  it("renders the inner div with the correct classes", () => {
    const { container } = render(<Header />);
    const innerDiv = container.querySelector(".d-flex.w-75.align-items-center.position-relative");

    expect(innerDiv).toBeInTheDocument();
  });

  it("renders the LogoSVG component", () => {
    const { getByTestId } = render(<Header />);
    const logoSVG = getByTestId("logo-svg");

    expect(logoSVG).toBeInTheDocument();
    expect(LogoSVG).toHaveBeenCalledTimes(3);
  });
});



describe("FormView Component", () => {
  it("renders the BackgroundContainer with the correct imageURL", () => {
    render(<FormView />);

    const backgroundContainer = screen.getByTestId("background-container");
    expect(BackgroundContainer).toHaveBeenCalledWith(
      expect.objectContaining({ imageURL: bgPhoto }),
      expect.anything()
    );
    expect(backgroundContainer).toBeInTheDocument();
  });

  it("renders the main structure and text", () => {
    render(<FormView />);

    const mainDiv = screen.getByText("Book amazing restaurants");
    const descriptionText = screen.getByText(
      /From local favorites to the trending restaurants, enjoy without the wait./i
    );

    expect(mainDiv).toBeInTheDocument();
    expect(descriptionText).toBeInTheDocument();
    expect(mainDiv).toHaveClass("font-40 font-weight-700");
  });

  it("renders the FormSection component", () => {
    render(<FormView />);

    const formSection = screen.getByTestId("form-section");
    expect(formSection).toBeInTheDocument();
    expect(formSection).toHaveTextContent("Mock FormSection");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<FormView />);
    expect(asFragment()).toMatchSnapshot();
  });
});

