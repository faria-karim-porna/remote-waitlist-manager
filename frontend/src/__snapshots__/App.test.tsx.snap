// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormView Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="background-container"
  >
    <div
      class="d-flex justify-content-center h-100"
    >
      <div
        class="w-75 d-flex align-items-center h-100"
      >
        <div
          class="form-view-main"
        >
          <div>
            <div
              class="font-40 font-weight-700"
            >
              Book amazing restaurants
            </div>
            <div
              class="my-4"
            >
              <div
                class="font-22 font-primary"
              >
                From local favorites to the trending restaurants, enjoy without the wait. Discover popular restaurants where you can skip the wait by booking through TableCheck.
              </div>
            </div>
          </div>
          <div
            data-testid="form-section"
          >
            Mock FormSection
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
